src/internal/serviceenv/option.go

```diff
@@ -1,52 +1,54 @@
 package serviceenv
 
 import (
-	"os"
-	"path/filepath"
-
-	"github.com/pachyderm/pachyderm/v2/src/internal/obj"
-	"github.com/pachyderm/pachyderm/v2/src/internal/storage/chunk"
-	"github.com/pachyderm/pachyderm/v2/src/internal/storage/fileset"
 	"github.com/pachyderm/pachyderm/v2/src/internal/storage/kv"
-	"github.com/pachyderm/pachyderm/v2/src/internal/uuid"
 )
 
-// ChunkStorageOptions returns the chunk storage options for the service environment.
-func (conf *Configuration) ChunkStorageOptions() ([]chunk.StorageOption, error) {
-	var opts []chunk.StorageOption
-	if conf.StorageUploadConcurrencyLimit > 0 {
-		opts = append(opts, chunk.WithMaxConcurrentObjects(0, conf.StorageUploadConcurrencyLimit))
+// ChunkMemoryCache returns the in memory cache for chunks, pre-configured to the desired size
+func (conf *Configuration) ChunkMemoryCache() kv.GetPut {
+	size := conf.StorageMemoryCacheSize
+	if size < 1 {
+		size = 1
 	}
-	if conf.StorageDiskCacheSize > 0 {
```

Comment:
What happened to the disk cache?

Reply:
This was moved into `chunk/option.go` to fix a circular dependency.

Reply:
And the upload limiter?